source/connection-troubleshooting.txt

```diff
@@ -0,0 +1,325 @@
+
+In most operating systems, each connection is associated with a `file
+descriptor
+<https://www.computerhope.com/jargon/f/file-descriptor.htm>`_. There is
+typically a limit set by the operating system on the number of file
+descriptors used by a single process. An ``ECONNRESET`` error can occur
+if the connection pool size surpasses the limit of ``file descriptors``.
+
+Consider the following operation:
+
+.. literalinclude:: /code-snippets/faq/econnresetWithClientConnect-example.js
+  :language: javascript
+  :linenos:
+
+If this operation causes an ``ECONNRESET`` error, you may have run into
+the ``file descriptor`` limit for your Node.js process. In that case you
+must increase the number of ``file descriptors`` for the Node.js process. On
+MacOS and Linux you do this with the `ulimit
+<https://ss64.com/bash/ulimit.html>`_ shell command.
+
+.. code-block:: sh
+
+   ulimit -n 6000
+
+This sets the maximum number of ``file descriptors`` for the process to
+6000, allowing Node.js to connect with a pool size of 5000 sockets.
+
+Authentication Error
+~~~~~~~~~~~~~~~~~~~~
+
+The {+driver-short+} can fail to connect to a MongoDB instance if
+the authorization is not configured correctly. In these cases, the driver raises
+an error message similar to one of the following messages:
+
+.. code-block:: none
+   :copyable: false
+
+   Command failed with error 18 (AuthenticationFailed): 'Authentication
+   failed.' on server localhost:27017.
+
+.. code-block:: none
+   :copyable: false
+
+   connection() error occurred during connection handshake: auth error:
+   sasl conversation error: unable to authenticate using mechanism
+   "SCRAM-SHA-256": (AuthenticationFailed) Authentication failed.
+
+The following sections describe methods that may help resolve the issue.
+
+.. _node-troubleshooting-connection-string-auth:
+
+Check Connection String
+-----------------------
+
+An invalid connection string is the most common cause of authentication
+issues when attempting to connect to MongoDB.
+
+.. tip::
+
+   For more information about connection strings,
+   see :ref:`Connection URI <node-connection-uri>` in the Connection Guide.
+
+If your connection string contains a username and password, ensure that they
+are in the correct format.
+
+.. note::
+
+   If the username or password includes any of the following characters, they
+   must be `percent encoded <https://tools.ietf.org/html/rfc3986#section-2.1>`__:
+
+   .. code-block:: none
+
+      : / ? # [ ] @
+
+When connecting to a replica set, you should include all of the hosts
+in the replica set in your connection string. Separate each of the hosts
+in the connection string with a comma. This enables the driver to establish a
+connection if one of the hosts is unreachable.
+
+.. _node-troubleshooting-connection-admin:
+
+Verify User Is in Authentication Database
+-----------------------------------------
+
+To successfully authenticate a connection by using a username and password,
```

Comment:
Suggestion:
I think it would be helpful to link to info on how to check for a user's permissions in the authentication database or perhaps a sample query. Otherwise, the user doesn't have a good path forward to checking whether their user is in the default admin database.
Also could be good to mention that connecting to a locally-hosted MongoDB deployment from the same machine doesn't require any auth so that, for those situations, the user could eliminate this as a potential solution.

Reply:
Due to another comment, I added a section about verifying the users permissions that links to the RBAC docs. This is specifically about making sure the user is in the right database, because that is a common issue based on forum hits.